Formula/mesos.rb

```diff
@@ -53,6 +53,16 @@ class Mesos < Formula
   def install
     ENV.java_cache
 
+    # Disable optimizing as libc++ does not play well with optimized clang
+    # builds (see https://llvm.org/bugs/show_bug.cgi?id=28469 and
+    # https://issues.apache.org/jira/browse/MESOS-5745).
+    #
+    # NOTE: We cannot use `--disable-optimize` since we also pass e.g.,
+    # CXXFLAGS via environment variables. Since compiler flags are passed via
+    # environment variables the Mesos build will silently ignore flags like
+    # `--[disable|enable]-optimize`.
+    ENV.O0
```

Comment:
@MikeMcQuaid: I have updated the patch to include a `TODO` to make this conditional once we can be certain the clang-9.0.0 and later fix this libcxx issue.

@ilovezfs: re:degrade gracefully in upstream, there seems to be little support for this since it would e.g., lead to automatic builds passing, but not using explicitly passed flags (e.g., `-O2` would be override with `-O0` in some cases). This is an issue for e.g., automated builds. I would like to suggest keeping an unconditional `-O0` here, and making it unconditional once clang-9.0.0 and the associated libcxx is released. Alternatively we could check here in Homebrew if a build with implicit `-O2` configures, and only drop to `-O0` if it does not -- this looks too complicated for my taste.